.github/workflows/clean-repo.yml

```diff
@@ -0,0 +1,42 @@
+name: "clean repo"
+
+on:
+  schedule:
+    - cron: "0 1 1 * *" # Runs at 01:00, on day 1 of the month
+
+jobs:
+  clean-repo:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      issues: write
+      pull-requests: write
+
+    steps:
+      - uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c
+
+      # Call clean repo
+      - name: Clean repo
+        id: clean-repo-step
+        uses: dotnet/docs-tools/cleanrepo/CleanRepo@main
+        with:
+          start-directory: '.'
+          docset-root: './docs'
+          repo-root: '.'
+          delete: true
+          orphaned-topics: true
```

Comment:
@IEvangelist Does it call the tool once with all of these arguments? If so, we'll need probably need to call it separately for each function, or I'll have to change the logic in the tool itself to call multiple functions successively.

Reply:
Yeah, whenever the action workflow is invoked (one time per month), it wakes up and calls your CleanRepo.dll once with all these switches. If you need help updating the app to accept multiple switches, I could help, that might be a more straightforward update. Otherwise, we'd end up re-creating the container for each call which is resource intensive.

Reply:
We could add a bash script to execute the following:
1. Build cleanrepo
2. iterate and call dotnet run with each of the switches in turn

Reply:
I'm not sure how that would work in this scenario and calling a bash script (while possible) differs from calling the CleanRepo GitHub Action. The CleanRepo tool is now exposed as a GitHub Action, via the _Dockerfile_. The image is built on demand, and the container is executed through the _action.yml_ - where does the bash script come into play?

Reply:
I didn't explain what I was thinking well enough.

Suppose the entry point in the action was the shell script. 

The first seven lines of the [Dockerfile](https://github.com/dotnet/docs-tools/blob/main/cleanrepo/CleanRepo/Dockerfile#L1-L7) would remain the same.

The remainder would be something like:

```txt
FROM mcr.microsoft.com/dotnet/sdk:7.0
COPY --from=build-env /app/out .
# Assume the build environment has a shell script named `clean-repo.sh`
ENTRYPOINT [ "/bin/sh", "/app/clean-repo.sh" ]
```

The `clean-repo.sh` script runs `dotnet /CleanRepo.dll` repeatedly with the different arguments.

Reply:
Ah, ok. That makes more sense now. Thank you for the added context. Let's ask @gewarren for her thoughts and preferences on this. I'm happy to try to update this when we hear back from her.

Reply:
Yes, let's try to loop through here rather than in the app itself.